src/content/contribute/writing-a-scaffold.md

```diff
@@ -0,0 +1,400 @@
+
+
+```js
+const Generator = require('yeoman-generator');
+const List = require('@webpack-cli/webpack-scaffold').List;
+
+module.exports = class WebpackGenerator extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
+    opts.env.configuration = {
+      dev: {}
+    };
+  }
+
+  prompting() {
+    return this.prompt([
+      List('confirm', 'Welcome to the demo scaffold! Are you ready?', ['Yes', 'No', 'Pengwings'])
+    ]).then (answer => {
+      if(answer['confirm'] === 'Pengwings') {
+        //
+      }
+    });
+  }
+};
+```
+
+
+## Configuring Webpack
+
+So far, we've made an interaction with the user. If you were coding along, great! So how do we proceed from here? Let's try to build a simple webpack config that has an entry point, an output, and a context property. For this, we need to create a `webpackOptions` property on our `dev` object. This is where `entry`, `output` and `context` is gonna be hooked up, later resulting in a `webpack.config.js`.
+
+T> Define the `webpackOptions` property in the constructor to make your scaffold as most clean as possible!
+
+```js
+const Generator = require('yeoman-generator');
+const List = require('@webpack-cli/webpack-scaffold').List;
+
+module.exports = class WebpackGenerator extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
+    opts.env.configuration = {
+      dev: {
+        webpackOptions: {}
+      }
+    };
+  }
+
+  prompting() {
+    return this.prompt([
+      List('confirm', 'Welcome to the demo scaffold! Are you ready?', ['Yes', 'No', 'Pengwings'])
+    ]).then (answer => {
+      if(answer['confirm'] === 'Pengwings') {
+        //
+      }
+    });
+  }
+};
+```
+
+
+## Dev Configs
+
+Congrats, you've now created the base of an `webpack-scaffold`! Let's add some stuff to our future configuration file!
+I'm going to follow good convention, and extract my config into another file, named `dev-config.js`. As this is just regular JavaScript, we can make the module a function, and supply our entry as a parameter for us to build up a configuration file from.
+
+__dev-config.js__
+
+```js
+module.exports = function createDevConfig(answer) {
+  let devConfig = {};
+};
+```
+
+__generator.js__
+
+```js
+const Generator = require('yeoman-generator');
+const List = require('@webpack-cli/webpack-scaffold').List;
+const createDevConfig = require('./dev-config');
+
+module.exports = class WebpackGenerator extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
+    opts.env.configuration = {
+      dev: {
+        webpackOptions: {}
```

Comment:
will you provide the example of what they have to write here for entry, output and context? to nicely follow the guide...